src/api/code-challenge/content-types/code-challenge/lifecycles.js

```diff
@@ -0,0 +1,92 @@
+module.exports = {
+  beforeCreate(event) {
+    console.log("**** beforeCreate hook is activated! ****");
+    const eventData = event.params.data;
+    runMetaTest(eventData);
+  },
+
+  beforeCreateMany(event) {
+    console.log("**** beforeCreateMany hook is activated! ****");
+    const eventData = event.params.data;
+    runMetaTest(eventData);
+  },
+
+  beforeUpdate(event) {
+    console.log("**** beforeUpdate hook is activated! ****");
+    const eventData = event.params.data;
+    runMetaTest(eventData);
+  },
+
+  beforeUpdateMany(event) {
+    console.log("**** beforeUpdateMany hook is activated! ****");
+    const eventData = event.params.data;
+    runMetaTest(eventData);
+  },
+};
+
+async function runMetaTest(eventData) {
+  const tests = await getInternalTests(eventData);
+  const metaTests = await getMetaTests(eventData);
+
+  tests.forEach((test) =>
+    console.log("final test.internalTest: ", test.internalTest)
+  );
+
+  metaTests.forEach((metaTest) =>
+    console.log("final metaTest caseCode: ", metaTest.caseCode)
+  );
+
+  return false;
+}
+
+async function getInternalTests(eventData) {
+  // Original Finding Internal Test
+  const internalTests = await strapi.db
+    .query("challenge.code-challenge-test")
+    .findMany({
+      select: ["id", "internalTest"],
+      where: {
+        id: {
+          $gte: getCurrentTestID(eventData),
+        },
+      },
+    });
+
+  console.log("internalTests: ", internalTests);
+  return internalTests;
+}
+
+function getCurrentTestID(eventData) {
+  let testIDs = [];
+  eventData.tests.forEach((test) => {
+    testIDs.push(test.id);
+  });
+  const currentTestID = Math.min(...testIDs);
```

Comment:
or actually use reduce that's even better

Reply:
Then the entire function can be done in one line

Reply:
although up to you, reduce will be fancy but maybe slightly less readable. Fancy can be fun though as long as it's not too tricky

Reply:
I changed it to forEach to map. I'm going to leave it that way for now. I might have to tweak things before the feature is complete.